src/OutputReaders/field_time_series_indexing.jl

```diff
@@ -227,6 +229,18 @@ end
 ##### FieldTimeSeries updating
 #####
 
+# Let's make sure `times` is available on the CPU. This is always the case 
+# for ranges. if `times` is a vector that resides on the GPU, it has to be moved to the CPU  for safe indexing. 
+# Copying the whole array is a bit unclean but considering `update_field_time_series!` is called quite rarely 
+# it should not impact too much memory / performance
```

Comment:
This is an assumption. We have no idea how often `update_field_time_series!` is called.

Reply:
You mean to say "for the simulations that I personally want to run".

I think you should rephrase this comment to make it clear that there is a penalty involved here, and we choose a method based on a certain assumption, and perhaps propose a TODO to fix it if use cases emerge that require it.